examples/sparse/sign.py

```diff
@@ -0,0 +1,82 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.data import CoraGraphDataset
+from dgl.mock_sparse import create_from_coo, diag, identity
+from torch.optim import Adam
+
+
+def sign(A, X, num_hops):
+    X_sign = [X]
+    for _ in range(num_hops):
+        X = A @ X
+        X_sign.append(X)
+    return X_sign
+
+
+class Model(nn.Module):
+    def __init__(self, in_size, num_hops, num_classes, hidden_size=256):
+        super().__init__()
+
+        num_feats = num_hops + 1
+        self.linear = nn.ModuleList(
+            [nn.Linear(in_size, hidden_size) for _ in range(num_feats)]
+        )
+        self.pred = nn.Linear(hidden_size * num_feats, num_classes)
+
+    def forward(self, X):
+        X = torch.cat([self.linear[i](X[i]) for i in range(len(X))], dim=1)
+        return self.pred(F.relu(X))
+
+
+def evaluate(X, Y, model):
+    model.eval()
+    with torch.no_grad():
+        logits = model(X)
+        _, Y_pred = torch.max(logits, dim=1)
+        return (Y_pred == Y).float().mean().item()
+
+
+dev = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+data = CoraGraphDataset()
+g = data[0].to(dev)
+src, dst = g.edges()
+N = g.num_nodes()
+A = create_from_coo(dst, src, shape=(N, N))
```

Comment:
In the paper W was used to represent adjacency matrix.
I am not sure about whether the community convention is to use A for adjacency matrix.
Here are 2 suggestions I would have.
1. Use the annotation in the paper directly in our code, and add a comment say, this is not dgl convention, but we are follow the paper.
2. Use the convention, but add a comment to clarify the annotation in the paper was W.

Reply:
It has been the standard to denote the adjacency matrix by A in graph theory, see https://en.wikipedia.org/wiki/Adjacency_matrix. The SIGN paper actually also used it in various spaces like equation (4) in https://arxiv.org/pdf/2004.11198.pdf.

Reply:
In the paper A was defined as A = D−1/2WD−1/2. My understand is, A in equation (4) also means this.

My feeling is, if we also confuses with the annotation, then it would happen to many other readers.
If using A is standard, then add a comment to point out instead of A, W was used in the paper for the adjacency matrix should be enough,

Reply:
It's often to see inconsistencies in notations across papers. For example, the symmetric adjacency matrix normalization was probably first known by many people with the [GCN paper](https://arxiv.org/abs/1609.02907), which is way more impactful than the SIGN paper. I follow its notations in naming variables.